internal/kafka/internal/services/cloud_providers_test.go

```diff
@@ -485,10 +498,61 @@ func Test_ListCloudProviderRegions(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "An error is returned when the cloud provider cannot be obtained",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return nil, errors.New("failed to get provider implementation")
+					},
+				},
+			},
+			wantErr: true,
+			setupFn: func() {
+				mocket.Catcher.Reset()
+				mocket.Catcher.NewMock().
+					WithQuery("SELECT DISTINCT").
+					WithReply([]map[string]interface{}{{"provider_type": "ocm"}, {"provider_type": "standalone"}})
+			},
```

Comment:
if it doesn't match received query, mocket simply exits without any error. That can lead in undetected errors.
Always add
```go
mocket.Catcher.NewMock().WithQueryException().WithExecException()
```
as latest catcher